### PR TITLE
Refactor skr-aws-upgrade-integration-dev test - part 2

### PR DESCRIPTION
**Pull request by hanngos**
**Description**
Refactor `skr-aws-upgrade-integration-dev` test

Changes proposed in this pull request:

- Wrap upgrade function, as it is in case of provisioning and deprovisioning
- Add new function `getOrProvisionSKR`
- Remove `commerce mock` and `compass`
- Fix `[Object object] log`
- Fix `dashboard_url`
- Add before and after blocks and wrap the content of the after block to a function
- Refactor OIDC to `upgrade` - this one will be added to follow-up PR

**Related issue(s)**
See also #14721

### CODE REVIEW

**Review by ralikio — DISMISSED** [review]

File: tests/fast-integration/skr-aws-upgrade-integration/index.js [hunk 1]
```diff
@@ -1,254 +1,64 @@
...
-  const keb = new KEBClient(KEBConfig.fromEnv());
-  const gardener = new GardenerClient(GardenerConfig.fromEnv());
+  switchDebug(true);
 
-  const kymaVersion = getEnvOrThrow('KYMA_VERSION');
-  const kymaUpgradeVersion = getEnvOrThrow('KYMA_UPGRADE_VERSION');
+  if (!skipProvisioning) {
+    globalTimeout += provisioningTimeout + deprovisioningTimeout; // 3h
+  }
+  this.timeout(globalTimeout);
```
Comment (szwedm) [inline tests/fast-integration/skr-aws-upgrade-integration/index.js:30]: trifle: overwritten in line 45

**Review by szwedm — APPROVED** [review]
/lgtm

File: tests/fast-integration/skr-test/README.md [hunk 2]
```diff
@@ -17,27 +14,22 @@ You can use the SKR test in two modes - with and without provisioning.
 
 In this mode, the test executes the following steps:
 
-1. Provision SKR cluster and register compass resources
+1. Provision SKR cluster
```
Comment (klaudiagrz) [inline tests/fast-integration/skr-test/README.md:17]:
```suggestion
1. Provision an SKR cluster.
```

File: tests/fast-integration/skr-test/README.md [hunk 3]
```diff
@@ -17,27 +14,22 @@ You can use the SKR test in two modes - with and without provisioning.
...
 2. Run OIDC Test
```
Comment (klaudiagrz) [inline tests/fast-integration/skr-test/README.md:18]:
```suggestion
2. Run the OIDC test.
```

File: tests/fast-integration/skr-test/README.md [hunk 4]
```diff
@@ -17,27 +14,22 @@ You can use the SKR test in two modes - with and without provisioning.
...
 - `SKIP_PROVISIONING`, set to `true`
 - `INSTANCE_ID` the uuid of the provisioned SKR instance
-- `TEST_SUFFIX` determines the compass scenario and the app assigned to it. If not set, it's randomly generated.
-
->**IMPORTANT:** The `TEST_SUFFIX must be 4 characters long and is required for every commerce-mock test run. If the compass scenario with the given suffix already exists, the test will try to reuse it.
 
 In this mode, the test executes the following steps:
-1. Ensure SKR exists and register compass resources
+1. Ensure SKR exists
 2. Run OIDC Test
```
Comment (klaudiagrz) [inline tests/fast-integration/skr-test/README.md:29]:
```suggestion
2. Run the OIDC test.
```

**Review by klaudiagrz — DISMISSED** [review]

**Review by szwedm — APPROVED** [review]
/lgtm

**Review by szwedm — APPROVED** [review]
/lgtm

**Review by klaudiagrz — APPROVED** [review]

**Review by szwedm — APPROVED** [review]
/lgtm

**Pull request merged**